docs/website/docs/reference/tuning.md

```diff
@@ -0,0 +1,136 @@
+---
+icon: octicons/meter-16
+---
+
+# Tuning
+
+This page documents support for IREE dispatch tuning. The compiler supports
+both default and user-provided tuning specs (specifications) that override
+compiler heuristics that guide dispatch code generation. In our experience,
+tuning specs can provide meaningful speedup of model execution. For example, we
+achieved a ~10% improvement on the Stable Diffusion XL (SDXL) model with the
+MI300X GPU.
+
+## Tuning specs
+
+The default specs are shipped with the IREE compiler and are target-specific.
+We aim to provide default tuning specs that cover the most in-demand hardware
+and dispatches from most popular ML models, although we do not guarantee
+completeness.
+
+User-provided tuning specs are a mechanism that allows for users to get the
+best performance on custom models and hardware targets without having to modify
+the compiler source code or needlessly special-case compiler heuristics.
+
+Currently, the dispatch tuner that generates tuning specs is still experimental
+and hosted
+[in an external repo](https://github.com/nod-ai/shark-ai/tree/main/tuner). This
+document describes how to work with tuning specs generated by the SHARK Tuner
+or produced by manually, but it does not go into detail on how to generate
+these specs.
+
+## Flags
+
+The use of tuning specs in `iree-compile` is controlled with the following
+flags:
+
+* `--iree-codegen-enable-default-tuning-specs` -- enables or disables the
+  default tuning specs shipped with the compiler.
+* `--iree-codegen-tuning-spec-path` -- loads a user-specified tuning spec.
+* `--iree-codegen-dump-tuning-specs-to` -- dumps final tuning specs to a
+  directory or standard output.
```

Comment:
This could also be a table, to make the flag names and descriptions stand out more and be easier to copy/paste.

List:
![image](https://github.com/user-attachments/assets/9860918e-036a-4ddb-9013-de9132b6267b)

Markdown table, inline code blocks:
![image](https://github.com/user-attachments/assets/220402f5-37c1-44d6-b8dd-51f52c283d82)

HTML table, fenced code blocks (with copy buttons):
![image](https://github.com/user-attachments/assets/16da0d64-0646-4e4d-b19f-2e34b6bcbee5)

Not perfect either way 🤔, especially with the copy button overlapping the first code block

Code for the last one if you want it:
````
<table>
<tr><th>Flag</th><th>Description</th><tr>
<tr><td>

```
--iree-codegen-enable-default-tuning-specs
```

</td><td>Enables or disables the default tuning specs shipped with the compiler.</td></tr>
<tr><td>

```
--iree-codegen-tuning-spec-path
```

</td><td>Loads a user-specified tuning spec.</td></tr>
<tr><td>

```
--iree-codegen-dump-tuning-specs-to
```

</td><td>Dumps final tuning specs to a directory or standard output.</td></tr>
</table>
````

Reply:
Could do something like
```html
<tr><th width="55%">Flag</th><th>Description</th><tr>
```
to bump the column width so the copy button doesn't overlap 🤔 (or other styling with css)

![image](https://github.com/user-attachments/assets/4bb7778c-ae81-41ec-84c8-37d1b591892e)

Reply:
I'm going to keep this as-is, I like the simplicity of bullet points. I don't expect this to grow much, so I think that tables are a bit of an overkill especially since they don't look that much better.

Reply:
SGTM!